modules/osint/github_leaks.py

```diff
@@ -0,0 +1,132 @@
+"""
+OWASP Maryam!
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import concurrent.futures
+from bs4 import BeautifulSoup
+import json
+
+meta = {
+	'name': 'GitHub Leaks',
+	'author': 'Aman Singh',
+	'version': '0.1',
+	'description': 'Search your query in the GitHub and show the potentially leaked info.',
+	'sources': ('google', 'carrot2', 'bing', 'yippy', 'yahoo', 'millionshort', 'qwant', 'duckduckgo', 'github'),
+	'options': (
+		('query', None, True, 'Query string', '-q', 'store', str),
+		('limit', 1, False, 'Search limit(number of pages, default=1)', '-l', 'store', int),
+		('count', 50, False, 'Number of links per page(min=10, max=100, default=50)', '-c', 'store', int),
+		('engine', 'google,github', False, 'Engine names for search(default=google, github)', '-e', 'store', str),
+		('thread', 2, False, 'The number of engine that run per round(default=2)', '-t', 'store', int),
+	),
+    'examples': ('github_leaks-q <QUERY> -l 15 -e carrot2,bing,qwant --output',)
+}
+
+LINKS = []
+PAGES = ''
+REPO = []
+OUTPUT = {'links': {}}
+
+def checks(self, repo):
+	threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=16)
+	futures = (threadpool.submit(leaks, self, link) for link in repo)
+	for results in concurrent.futures.as_completed(futures):
+		print(f"Found {len(OUTPUT['links'])} links" , end= '\r')
+	print('\n')
+
+def leaks(self, repo):
+	checks = ['password','npmrc _auth','dockercfg','pem private','id_rsa','aws_access_key_id','s3cfg','htpasswd','git-credentials','bashrc password','sshd_config','xoxp OR xoxb OR xoxa','SECRET_KEY','client_secret','sshd_config','github_token','api_key','FTP','app_secret','passwd','s3.yml','.exs','beanstalkd.yml','deploy.rake','mysql','credentials','PWD','deploy.rake','.bash_history','.sls','secrets','composer.json']
```

Comment:
Check the advance search in github. we don't need to search all of them separately!
e.g: if these are keyword: _password OR npmrc _auth OR id_rsa_
if these are filename: `filename:composer.json filename:deploy.rake filename:s3.yml`
if these are extension: `extension:sls extension:exs`
Categorize them based on the filename, extension, keyword.
deploy.rake has been duplicated.

Reply:
this can't be done for keywords, rest done